### DISCUSSION TITLE
接口测速问题

### DISCUSSION THREAD

**Discussion by tianniao886** [Guovin/iptv-api#739] [Q&A] [answered]
Docker版ipv4、ipv6接口测速不稳定。如：生成的ipv4频道有的会卡顿，同一个频道ipv6接口没有速度，但观看不卡顿。是不是ipv6接口测速不准确？是否能优化下？
Name: CCTV-11, URL:  http://58.19.43.46:9901/tsfile/live/1030_1.m3u8 $订阅源, Date: None, Delay: 34 ms, Speed: 20.13 M/s, Resolution: None
Name: CCTV-11, URL:  http://[2409:8087:5e08:24::11]:6610/000000001000/1000000005000265011/1.m3u8?channel-id=ystenlive&Contentid=1000000005000265011&livemode=1&stbId=3 $订阅源, Date: None, Delay: -inf ms, Speed: inf M/s, Resolution: None
![屏幕截图 2024-12-25 120704](https://github.com/user-attachments/assets/42eb6bbb-96ed-46cc-86a9-4044676dc667)

**Comment by Guovin** [comment]
检查你的docker网络，应该是没有ipv6支持，导致无法对ipv6测速

**Comment by tianniao886** [comment]
docker host 模式有ipv6，某些ipv6接口有速度，显示speed：0.01-0.02m/s。上一版ipv4，ipv6接口测速都显示正常。

**Comment by Guovin** [comment]
日志显示ipv6的测速结果都是一样的，这只有不支持ipv6才能出现的情况，具体请给出完整的容器运行日志和sort.log

**Comment by tianniao886** [comment]
刚出锅的...麻烦大佬给看下吧..非常感谢！！
[iptv-api.txt](https://github.com/user-attachments/files/18245575/iptv-api.txt)
[sort.log](https://github.com/user-attachments/files/18245593/sort.log)

**Comment by tianniao886** [comment]
设置文件及生成的频道文件
[config.txt](https://github.com/user-attachments/files/18245564/config.txt)
[result.txt](https://github.com/user-attachments/files/18245565/result.txt)

**Comment by Guovin** [comment]
把你当前使用的镜像版本信息发一下？是不是最新版？如果你的网络支持ipv6，建议修改配置ipv6_support = True跳过支持检查。

**Comment by tianniao886** [comment]
好的
![屏幕截图 2024-12-26 102330](https://github.com/user-attachments/assets/32288d03-8fb8-4e76-a597-f09a8317b8bb)

**Comment by Guovin** [comment]
我看到你的配置sort_timeout = 2，有可能是时间过短，ipv6没有读取到资源就被中断，导致平均速率都普遍接近0，建议改回10，再运行一次试试

**Comment by tianniao886** [comment]
已修改config sort_timeout = 10 ，ipv6_support = True，其他参数未动。ipv6接口测速显示speed：0.00~0.01m/s。还是原来那样。
[2024.12.26sort.log](https://github.com/user-attachments/files/18249213/2024.12.26sort.log)
[2024.12.26result.txt](https://github.com/user-attachments/files/18249214/2024.12.26result.txt)

**Comment by tianniao886** [comment]
[2024.12.26iptv-api.zip](https://github.com/user-attachments/files/18249260/2024.12.26iptv-api.zip)
[2024.12.26config.txt](https://github.com/user-attachments/files/18249262/2024.12.26config.txt)
Docker日志及config.ini

**Comment by Guovin** [comment]
ok，我这边会去复现并解决。

**Comment by Guovin** [comment, marked as answer]
该问题已修复，请拉取最新镜像

**Reply by hktkjj** [reply]
大佬威武，一杯咖啡敬上

**Reply by Guovin** [reply]
@hktkjj 感谢支持~
